pkg/kubectl/resource/builder.go

```diff
@@ -290,28 +292,53 @@ func (b *Builder) ResourceNames(resource string, names ...string) *Builder {
 	return b
 }
 
-// SelectorParam defines a selector that should be applied to the object types to load.
+// LabelSelectorParam defines a selector that should be applied to the object types to load.
 // This will not affect files loaded from disk or URL. If the parameter is empty it is
 // a no-op - to select all resources invoke `b.Selector(labels.Everything)`.
-func (b *Builder) SelectorParam(s string) *Builder {
+func (b *Builder) LabelSelectorParam(s string) *Builder {
 	selector, err := labels.Parse(s)
 	if err != nil {
-		b.errs = append(b.errs, fmt.Errorf("the provided selector %q is not valid: %v", s, err))
+		b.errs = append(b.errs, fmt.Errorf("the provided labelSelector %q is not valid: %v", s, err))
 		return b
 	}
 	if selector.Empty() {
 		return b
 	}
 	if b.selectAll {
-		b.errs = append(b.errs, fmt.Errorf("found non empty selector %q with previously set 'all' parameter. ", s))
+		b.errs = append(b.errs, fmt.Errorf("found non empty labelSelector %q with previously set 'all' parameter. ", s))
 		return b
 	}
-	return b.Selector(selector)
+	return b.LabelSelector(selector)
 }
 
-// Selector accepts a selector directly, and if non nil will trigger a list action.
-func (b *Builder) Selector(selector labels.Selector) *Builder {
-	b.selector = selector
+// LabelSelector accepts a labelSelector directly, and if non nil will trigger a list action.
+func (b *Builder) LabelSelector(selector labels.Selector) *Builder {
+	b.labelSelector = selector
+	return b
+}
+
+// FieldSelectorParam defines a selector that should be applied to the object types to load.
+// This will not affect files loaded from disk or URL. If the parameter is empty it is
+// a no-op - to select all resources invoke `b.Selector(fields.Everything)`.
+func (b *Builder) FieldSelectorParam(s string) *Builder {
+	selector, err := fields.ParseSelector(s)
```

Comment:
looks like this is still parsing, which is not really what we want

Reply:
> looks like this is still parsing, which is not really what we want

Will open another PR on this.

> So FieldSelector needs to be passed unmodified from the command line to the remote server.

@liggitt What about current `LabelSelector`? Making it the same with `FieldSelector`?

Reply:
ping @liggitt Done. PTAL. Thanks.

Open another PR #52971 for `labelSelector`. PTAL.